app/controllers/concerns/wrap_in_with_deleted.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+module WrapInWithDeleted
+  extend ActiveSupport::Concern
+
+  included do
+    around_action :wrap_in_with_deleted, if: proc {
+      request.get? && (with_deleted? || search_with_deleted?)
+    }
+  end
+
+  protected
+
+  def wrap_in_with_deleted
+    Project.with_deleted do
+      Stage.with_deleted do
+        Deploy.with_deleted do
+          yield
+        end
+      end
+    end
+  end
+
+  # For both deleted and non-deleted resources
+  def with_deleted?
+    params[:with_deleted] == "true"
+  end
+
+  def search_with_deleted?
+    params.dig(:search, :deleted) == "true"
```

Comment:
change the input to `with_deleted` and then get rid of this ?

Reply:
Fine, but we will miss the functionality of displaying the combined result of deleted and non-deleted deploys.
This combined results may help the user to view the deploys in exact order.
What's your thought on this functionality?

Reply:
ah, I did not think of that usecase :D